Add to_fingerprint_md5_string to get MD5 fingerprint

genedna opened the pull request:
Signed-off-by: Eli Ma <eli@patch.sh>
---
1. In some scenarios use the _*old*_ MD5 fingerprint to manipulate the SSH public key, so I add the `to_fingerprint_md5_string` to get. To get _*MD5*_ fingerprint with command `ssh-keygen -l -E md5 -f key`.
2. Re-implement the `fingerprint()` function with `rust-crypto`, and remove the `sha2` library.

Scenarios examples:
* destroy an SSH key in [Digitalocean API](https://developers.digitalocean.com/documentation/v2/#destroy-a-key)

Also, I added a test case and passed:

```
test tests::rsa_fingerprint_md5_string ... ok
```

lucab commented on Cargo.toml:
```diff
@@ -20,3 +20,4 @@ base64 = "0.6"
 byteorder = "1.1"
 error-chain = { version = "0.11", default-features = false }
 sha2 = "0.6"
+rust-crypto = "0.2.26"
```
Is the `.26` required because of some specific bugfix? If not, then please just cut it at "0.2".

genedna replied:
@lucab Yeah, it's not necessary. I updated the version to `^0.2` in the `75cf310` commit.

lucab commented:
One of the failures here was introduce sometime ago directly on master. https://github.com/sdemos/openssh-keys/pull/10 should take care of that.

sdemos commented:
sweet! thanks for the contribution. I noticed that this uses the `rust-crypto` crate. we should probably either switch this pr to the [`md-5`](https://docs.rs/md-5/0.7.0/md5/) crate (which is from the same family of hash crates) or use `rust-crypto` for the sha2 work as well, for consistency. I don't have any particularly strong feelings in either direction, so let's just move to `rust-crypto` everywhere. 

@genedna if you want to do that work and tack it onto this pr as well in a separate commit, that would be awesome. otherwise, let me know and I'll just switch it over sometime later today. 

also, unrelated, I merged #10 so go ahead and rebase and the travis error should go away.

genedna commented:
@sdemos ,

Honestly, I don't know what's different between `rust-crypto` **SHA256** function and `sha2`. It looks like `rust-crypto` more popular than `sha2`.

I replaced the `sha2` with `rust-crypto` in the https://github.com/genedna/openssh-keys/commit/b4f2621530b45d8b6e6d71702e04c7b4bf585621. The tests all pass and look all right.

**rust-crypto** stats overview
![screenshot-2017-11-30 rust-crypto - cargo packages for rust](https://user-images.githubusercontent.com/76467/33410583-e5e4a614-d5bb-11e7-91c3-a4bc716b3534.png)

**sha2** stats overview
![screenshot-2017-11-30 sha2 - cargo packages for rust](https://user-images.githubusercontent.com/76467/33410584-e650c60a-d5bb-11e7-8edb-ed2aa1fe141f.png)#

lucab commented:
rust-crypto refers to https://github.com/DaGenix/rust-crypto, which is pretty much a stale/dead project. I'd suggest to go with `md-5` and `sha2` instead.

sdemos commented:
hmm yeah that's a good point, I didn't notice that. let's switch it back to the `md-5` and `sha2` libraries instead. sorry for the flip flop @genedna!

genedna commented:
@sdemos ,

1. I replaced `rust-crypto` with `md-5` to implemented the md5 fingerprint function.
2. I split md5 function to `fingerprint_m5` and `to_fingerprint_md5_string`. The `fingerprint_md5` export _*fingerprint*_ like `fb:a0:5b:a0:21:01:47:33:3b:8d:9e:14:1a:4c:db:6d`, and `to_fingerprint_md5_string` export _*fingerprint*_ like `2048 MD5:fb:a0:5b:a0:21:01:47:33:3b:8d:9e:14:1a:4c:db:6d demos@anduin (RSA)` .

P.S.
@lucab The PR still has CI error, and I already `rebase` my fork repository with command,

```
git remote add upstream git@github.com:sdemos/openssh-keys.git
git fetch upstream
git rebase upstream/master
git push -f
```

sdemos commented:
by the way, this is now in the latest release (`v0.2.1`) so go ahead and update to that if you haven't already. thanks again!

genedna commented:
@sdemos , OK, I will try it.
